pkg/controller/controller.go

```diff
@@ -326,6 +330,13 @@ func (c *Controller) scaleNodeGroup(nodegroup string, nodeGroup *NodeGroupState)
 		}
 	}
 
+	// Special handling of pending pods when there's no running node.
+	// At this point, we know there's nothing we can do for the pending pods. So, let's scale up one node!
+	// If the newly scaled up node cannot handle all the pods, the existing scale up logic will kick in and handle the rest of the remaining pod.
+	if len(allNodes) == 0 && len(pods) > 0 {
+		nodesDelta = 1
```

Comment:
@Jacobious52 I did tried your suggested approach but there's an edge case where batch job doesn't provide any resource requirement, it was not possible to obtain cpu or mem. With this limitation, I went with the slower but safer approach.

Reply:
Good catch on that edge case. The reason we don't consider this as the design of escalator expects any pods that want to be considered in the calculation to have their resource requirements defined. So this should be a safe assumption, as escalator will not work correctly anyway if the resource requirements aren't defined on the pods. We enforce this requirement with an admission controller, however I don't think we are explicit about this in the docs.

Reply:
@Jacobious52 It's under the Calculations -> Assumptions section in the docs - https://github.com/atlassian/escalator/blob/master/docs/calculations.md#assumptions

Reply:
oh.. sorry didn't see that page before. I'm fine with these assumptions, I'll re-work on the logic then. Will you guys consider keeping this logic for just in case of the edge case?

Reply:
Yes you will still need a fallback scale up amount for instances when the node resource sizes are not known yet. One case for this would be starting Escalator with an ASG of size 0.